onnxscript/version_converter/version_converter.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from __future__ import annotations
+
+from onnxscript import ir
+
+### Adapters
+
+# Compatibility Adapter
+def adapter_compatible(op: ir.Node, target_opset):
+    op.version = target_opset
+
+
+# Adapter to convert axis from an attribute to a node input
+def adapter_axis_attr_to_input(op: ir.Node, target_opset):
```

Comment:
I suggest parameterizing this over the attribute name to make it reusable as below:
```py
def change_attr_to_input(attr_name: str) -> Adapter:
    def adapt (node: ir.Node):
        ...
    return adapt
```
This will let us register `change_attr_to_input("axis")` as an adapter, but also use it for other attributes as needed